### ISSUE TITLE
Error (Warning?) during cpanm

### ISSUE THREAD

**Issue by s1037989** [jhthorsen/app-git-ship#3]
$ cpanm App::git::ship
--> Working on App::git::ship
Fetching http://www.cpan.org/authors/id/J/JH/JHTHORSEN/App-git-ship-0.19.tar.gz ... OK
Configuring App-git-ship-0.19 ... OK
Scalar found where operator expected at (eval 112) line 1, near "*/$1"
    (Missing operator before $1?)
==> Found dependencies: Applify
--> Working on Applify
Fetching http://www.cpan.org/authors/id/J/JH/JHTHORSEN/Applify-0.12.tar.gz ... OK
Configuring Applify-0.12 ... OK
Building and testing Applify-0.12 ... OK
Successfully installed Applify-0.12
Building and testing App-git-ship-0.19 ... OK
Successfully installed App-git-ship-0.19
2 distributions installed

**Comment by jhthorsen** [comment]
No idea why this happens. Do you have any more input?

**Comment by s1037989** [comment]
Gosh, I wish I did.  That's a CnP from start to finish!  [Checkout this screencast](https://asciinema.org/a/e1osuc0amc8hybp9hod9vo6nr) showing just that!  :)

It installs and operates just fine, it just gives that weird warning upon doing so.

Let me know if there's anything more I can do to help.  BTW, this happened on a second box now.  Both are Ubuntu 14.04.4.

```
$ uname -a ; perl -E 'say $]'
Linux deskws 3.13.0-74-generic #118-Ubuntu SMP Thu Dec 17 22:52:02 UTC 2015 i686 i686 i686 GNU/Linux
5.018002
```

`cpanm`'s `build.log` looks good:

```
$ cat ~/.cpanm/build.log 
cpanm (App::cpanminus) 1.7001 on perl 5.018002 built for i686-linux-gnu-thread-multi-64int
Work directory is /home/stefan/.cpanm/work/1452146256.7336
You have make /usr/bin/make
You have LWP 6.05
You have /bin/tar: tar (GNU tar) 1.27.1
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
Checking if you have ExtUtils::MakeMaker 6.31 ... Yes (7.10)
Checking if you have ExtUtils::Install 1.46 ... Yes (1.59)
Searching App::git::ship on cpanmetadb ...
--> Working on App::git::ship
Fetching http://www.cpan.org/authors/id/J/JH/JHTHORSEN/App-git-ship-0.19.tar.gz
-> OK
Unpacking App-git-ship-0.19.tar.gz
Entering App-git-ship-0.19
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.10)
Configuring App-git-ship-0.19
Running Makefile.PL
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for App::git::ship
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Test::Pod::Coverage 0 ... Yes (1.10)
Checking if you have IPC::Run3 0.04 ... Yes (0.048)
Checking if you have Module::CPANfile 1.00 ... Yes (1.1001)
Checking if you have Module::Find 0.10 ... Yes (0.13)
Checking if you have ExtUtils::MakeMaker 6.00 ... Yes (7.10)
Checking if you have Test::More 0.90 ... Yes (1.001003)
Checking if you have CPAN::Uploader 0.10 ... Yes (0.103012)
Checking if you have Test::Pod 0 ... Yes (1.51)
Checking if you have Applify 0.06 ... Yes (0.12)
Checking if you have File::Path 2.00 ... Yes (2.09)
Building and testing App-git-ship-0.19
cp lib/App/git/ship/perl.pm blib/lib/App/git/ship/perl.pm
cp lib/App/git/ship.pm blib/lib/App/git/ship.pm
cp bin/git-ship blib/script/git-ship
"/usr/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/git-ship
Manifying 2 pod documents
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-basic.t ................ ok
t/config-with-comments.t .... ok
t/perl-basic.t .............. ok
# Workdir is workdir/perl-build
t/perl-build.t .............. ok
# Workdir is workdir/perl-changelogmd
# Init alien files
# Build
# Ship
t/perl-ship-changelog-md.t .. ok
# Workdir is workdir/perl-ship
# First release
# Second release
t/perl-ship.t ............... ok
# Workdir is workdir/perl-start
t/perl-start.t .............. ok
# Workdir is workdir/repository
t/repository.t .............. ok
t/ship-basic.t .............. ok
# Workdir is workdir/ship-start
t/ship-start.t .............. ok
t/silent.t .................. ok
All tests successful.
Files=11, Tests=128,  4 wallclock secs ( 0.06 usr  0.01 sys +  2.95 cusr  0.38 csys =  3.40 CPU)
Result: PASS
Manifying 2 pod documents
Installing /home/stefan/perl5/lib/perl5/App/git/ship.pm
Installing /home/stefan/perl5/lib/perl5/App/git/ship/perl.pm
Installing /home/stefan/perl5/man/man3/App::git::ship.3pm
Installing /home/stefan/perl5/man/man3/App::git::ship::perl.3pm
Installing /home/stefan/perl5/bin/git-ship
Appending installation info to /home/stefan/perl5/lib/perl5/i686-linux-gnu-thread-multi-64int/perllocal.pod
-> OK
Successfully installed App-git-ship-0.19
Installing /home/stefan/perl5/lib/perl5/i686-linux-gnu-thread-multi-64int/.meta/App-git-ship-0.19/install.json
1 distribution installed
```

**Comment by s1037989** [comment]
There's currently a bug in one of the tests of t/perl-start.t, so I commented out that block in my local master.  I then generated a dist with `make dist`.  I then installed `App::git::ship` just like above, except that I am pointing at a local tarball instead of to a Module name to be retrieved from CPAN.  I assume the method that `cpanm` uses is identical between the two.  If so, the only difference is that the failure above is from v0.19 and the success shown below is v0.20.

```
$ cpanm App-git-ship-0.20.tar.gz 
--> Working on App-git-ship-0.20.tar.gz
Fetching file:///tmp/app-git-ship/App-git-ship-0.20.tar.gz ... OK
Configuring App-git-ship-0.20 ... OK
==> Found dependencies: Applify
--> Working on Applify
Fetching http://www.cpan.org/authors/id/J/JH/JHTHORSEN/Applify-0.12.tar.gz ... OK
Configuring Applify-0.12 ... OK
Building and testing Applify-0.12 ... OK
Successfully installed Applify-0.12
Building and testing App-git-ship-0.20 ... OK
Successfully installed App-git-ship-0.20
2 distributions installed
```

I pulled down v0.19 and did the exact same method, and didn't get the error from above!

```
$ cpanm App-git-ship-0.19.tar.gz 
--> Working on App-git-ship-0.19.tar.gz
Fetching file:///tmp/app-git-ship-0.19/App-git-ship-0.19.tar.gz ... OK
Configuring App-git-ship-0.19 ... OK
==> Found dependencies: Applify
--> Working on Applify
Fetching http://www.cpan.org/authors/id/J/JH/JHTHORSEN/Applify-0.12.tar.gz ... OK
Configuring Applify-0.12 ... OK
Building and testing Applify-0.12 ... OK
Successfully installed Applify-0.12
Building and testing App-git-ship-0.19 ... OK
Successfully installed App-git-ship-0.19
2 distributions installed
```

I then installed `App::git::ship` per the original `cpanm App::git::ship` and got the same original error (and of course because the release on CPAN is v0.20, I get a build error, see issue #6):

```
$ rm -rf /home/stefan/perl5/lib/perl5/Applify.pm /tmp/app-git-ship/blib/lib/App/git
$ cpanm App::git::ship
--> Working on App::git::ship
Fetching http://www.cpan.org/authors/id/J/JH/JHTHORSEN/App-git-ship-0.20.tar.gz ... OK
Configuring App-git-ship-0.20 ... OK
Scalar found where operator expected at (eval 107) line 1, near "*/$1"
    (Missing operator before $1?)
==> Found dependencies: Applify
--> Working on Applify
Fetching http://www.cpan.org/authors/id/J/JH/JHTHORSEN/Applify-0.12.tar.gz ... OK
Configuring Applify-0.12 ... OK
Building and testing Applify-0.12 ... OK
Successfully installed Applify-0.12
Building and testing App-git-ship-0.20 ... FAIL
! Installing App::git::ship failed. See /home/stefan/.cpanm/work/1452668322.6509/build.log for details. Retry with --force to force install it.
1 distribution installed
```

Oh, and v0.20 generates the same error when installing from CPAN!

So, in summary, v0.19 and v0.20 both have this weird warning with `cpanm App::git::ship` but neither have the weird warning with `cpanm App-git-ship.tar.gz`.

I'm really not sure how to go about debugging this, so I think I'll clone App::git::ship, change the namespace, and upload it to CPAN for testing purposes, then I'll strip the module down to it's barest components and go from there.

**Comment by jhthorsen** [comment]
Yeah. The bug is #8. Going to fix it now.

**Comment by jhthorsen** [comment]
Fixed failing t/perl-start.t in 5e8fd3b.

**Comment by jhthorsen** [comment]
Not sure if this is interesting, but I just installed git-ship:

```
$ cpanm App::git::ship
--> Working on App::git::ship
Fetching http://www.cpan.org/authors/id/J/JH/JHTHORSEN/App-git-ship-0.21.tar.gz ... OK
Configuring App-git-ship-0.21 ... OK
Building and testing App-git-ship-0.21 ... OK
Successfully installed App-git-ship-0.21 (upgraded from 0.19)
1 distribution installed
$ perl -V | head -n1
Summary of my perl5 (revision 5 version 16 subversion 0) configuration:
$ cpanm --version
cpanm (App::cpanminus) version 1.5013
```

**Comment by s1037989** [comment]
I mentioned above that two different boxes caused this problem: one was cpanm version 1.7001 and the other version 1.7009.  I updated cpanm to the latest 1.7040 and the weird warning message went away!

**Comment by jhthorsen** [comment]
Cool! :+1:
